src/coord/src/coord.rs

```diff
@@ -1200,6 +1200,25 @@ impl Coordinator {
             .filter(|(_, frontier)| frontier != &Antichain::new())
             .collect();
 
+        for (id, frontier) in since_updates.iter() {
+            let item = self.catalog.try_get_by_id(*id).map(|e| e.item());
+            if let Some(CatalogItem::Index(catalog::Index { on, .. })) = item {
+                if let CatalogItem::Table(catalog::Table {
+                    persist: Some(persist),
+                    ..
+                }) = self.catalog.get_by_id(on).item()
+                {
+                    if self.catalog.default_index_for(*on) == Some(*id) {
+                        // TODO: more idiomatic antichain usage.
```

Comment:
let's rebase this on top of my recent PR and resolve this TODO (and also use multiwriteahandle to make this into one persist call per maintenance call)

Reply:
resolved the todo. I did not change this to use multiwritehandle at this time because the api requires a list of (Id, Frontier) pairs and we would need a map from globalid -> id on PersistMultiDetails (or something like that) to make this work if I understand correctly.

It also seemed to me like the perf win from batching up multiple allow_compaction messages into one wouldn't be that big even with many tables because the batched message does essentially the same work as a run of unbatched messages (modify the in memory state and wait to get flushed by drain_pending) and the batched message itself is not dramatically smaller (unlike the seal api where we can get away with a list of ids and a single seal frontier). In this case, the batched message is likely approximately the same size as the sum of the unbatched message sizes minus some constant overhead.

Given that plus the additional complexity of changing PersistMultiWrites I felt it was ok to punt on the multi-write-handle work for allow_compaction. wdyt?

Reply:
I'd rather do this now. You should be able to call `persist.write_handle.stream_id()` to get the persist `Id`

Reply:
aha! forgot about that function. yup done